src/plugins/discover/public/application/view_components/utils/use_prometheus.ts

```diff
@@ -28,7 +29,7 @@ export const usePrometheus = (): PrometheusContext => {
     services: { data },
   } = useOpenSearchDashboards<DiscoverViewServices>();
   const prometheusResourceClient = useMemo(() => {
-    return data.resourceClientFactory.create('prometheus');
+    return data.resourceClientFactory.get('prometheus') as PrometheusResourceClient;
```

Comment:
Moving forward we can figure make sure the Prometheus client factory gets vended properly in this situation. Happy to sync with you on suggestions here.